### PR TITLE
Linux paths

### PR DESCRIPTION
**Pull request by benjaminc**
Add support for linux paths to the IOHelper, and update the MapPath callers to instead call the IHostingEnvironment methods which distinguish between content and web roots.

### CODE REVIEW

File: src/Umbraco.Core/Configuration/XmlConfigManipulator.cs [hunk 1]
```diff
@@ -5,25 +5,25 @@
...
-            _ioHelper = ioHelper;
+            _hostingEnvironment = hostingEnvironment;
             _logger = logger;
         }
 
         public void RemoveConnectionString()
         {
             var key = Constants.System.UmbracoConnectionName;
-            var fileName = _ioHelper.MapPath("~/web.config");
+            var fileName = _hostingEnvironment.MapPathContentRoot("~/web.config");
```
Comment (JimBobSquarePants) [inline src/Umbraco.Core/Configuration/XmlConfigManipulator.cs:26]: Why is `web.config` still a thing?

Comment (bergmania) [reply src/Umbraco.Core/Configuration/XmlConfigManipulator.cs]: It's a legacy thing. I removed it.

File: src/Umbraco.Tests.Common/TestHelperBase.cs [hunk 2]
```diff
@@ -86,7 +87,11 @@ public IIOHelper IOHelper
             get
             {
                 if (_ioHelper == null)
-                    _ioHelper = new IOHelper(GetHostingEnvironment());
+                {
+                    var hostingEnvironment = GetHostingEnvironment();
+                    _ioHelper = RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? new IOHelperLinux(hostingEnvironment)
+                        : (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? (IIOHelper)new IOHelperOSX(hostingEnvironment) : new IOHelperWindows(hostingEnvironment));
```
Comment (rustybox) [inline src/Umbraco.Tests.Common/TestHelperBase.cs:93]: That's a lot of ternaries, would read a lot nicer if it wern't.

Comment (JimBobSquarePants) [reply src/Umbraco.Tests.Common/TestHelperBase.cs:93]: I would personally create a dedicated class for environmental properties. `TestEnvironment`

``` c#
public static bool IsLinux => RuntimeInformation.IsOSPlatform(OSPlatform.Linux);

public  static bool IsOSX => RuntimeInformation.IsOSPlatform(OSPlatform.OSX);

public  static bool IsWindows => RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
```

**Pull request merged**